test/shared/WalletClient.ts

```diff
@@ -0,0 +1,303 @@
+import Client, { CLIENT_EVENTS } from "@walletconnect/client";
+import { ClientOptions, IClient, SessionTypes } from "@walletconnect/types";
+import { ERROR } from "@walletconnect/utils";
+import { SIGNER_EVENTS } from "@walletconnect/signer-connection";
+import { formatJsonRpcError, formatJsonRpcResult } from "@walletconnect/jsonrpc-utils";
+import {
+  NodeProvider,
+  SignTransferTxParams,
+  SignDeployContractTxParams,
+  SignExecuteScriptTxParams,
+  SignUnsignedTxParams,
+  SignHexStringParams,
+  SignMessageParams,
+  Account,
+} from "alephium-web3";
+import { PrivateKeyWallet } from "alephium-web3/test";
+
+import WalletConnectProvider, {
+  isCompatibleChainGroup,
+  parseChain,
+  formatChain,
+  formatAccount,
+  providerEvents,
+} from "../../src";
+
+export interface WalletClientOpts {
+  privateKey: string;
+  networkId: number;
+  rpcUrl: string;
+  submitTx?: boolean;
+}
+
+export type WalletClientAsyncOpts = WalletClientOpts & ClientOptions;
+
+export class WalletClient {
+  public provider: WalletConnectProvider;
+  public nodeProvider: NodeProvider;
+  public signer: PrivateKeyWallet;
+  public networkId: number;
+  public rpcUrl: string;
+  public submitTx: boolean;
+
+  public client?: IClient;
+  public topic?: string;
+
+  public permittedNetworkId?: number;
+  public permittedChainGroup?: number;
+
+  get permittedChain(): string {
+    if (
+      typeof this.permittedNetworkId === "undefined" ||
+      typeof this.permittedChainGroup === "undefined"
```

Comment:
Is there a specific reason as to why checking the `typeof` instead of simply doing
```ts
     this.permittedNetworkId === undefined ||
     this.permittedChainGroup === undefined
```
?

Reply:
`typeof` is recommended as far as I searched: https://thisthat.dev/variable-undefined-vs-typeof-variable-undefined/

Reply:
OK, redefining global `undefined` is a very old browser issue, I wouldn't consider it a problem anymore. Also, typescript doesn't let us use undeclared vars, so I wouldn't consider that a problem either.

Reply:
If we are sure that using `=== undefined` is enough, I will be happy to change. I hate it everytime I type `typeof foo === 'undefined'`

Reply:
Will make this change to all my code 💯

Reply:
Just checked the official `walletconnect` repo and `polkajs-common`, and both are still using `typeof`.

Is it a legacy thing, or I am missing anything?

Reply:
I'll investigate further soon, at the moment it seems to me that it's a legacy thing.